spec/Plugin/BaseUriPluginSpec.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace spec\Http\Client\Common\Plugin;
+
+use Http\Message\StreamFactory;
+use Http\Message\UriFactory;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use PhpSpec\ObjectBehavior;
+
+class BaseUriPluginSpec extends ObjectBehavior
+{
+    function let(UriInterface $uri)
+    {
+        $this->beConstructedWith($uri);
+    }
+
+    function it_is_initializable(UriInterface $uri)
+    {
+        $uri->getHost()->shouldBeCalled()->willReturn('example.com');
```

Comment:
I tried this:

```php
    function let(UriInterface $uri)
    {
        $uri->getPath()->shouldBeCalled()->willReturn('/api');

        $this->beConstructedWith($uri);
    }
```

But got this error:

```

        Http/Client/Common/Plugin/AddPathPlugin
  47  ✘ throws exception on trailing slash
        some predictions failed:
          Double\UriInterface\P114:
            No calls have been made that match:
              Double\UriInterface\P114->getPath()
            but expected at least one.

        Http/Client/Common/Plugin/AddPathPlugin
  56  ✘ throws exception on empty path
        some predictions failed:
          Double\UriInterface\P116:
            No calls have been made that match:
              Double\UriInterface\P116->getPath()
            but expected at least one.
```

I don't think this can be refactored on the `let` method because the return value changes on some cases.